dali/pipeline/graph/graph2dot.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_GRAPH_GRAPH2DOT_H_
+#define DALI_PIPELINE_GRAPH_GRAPH2DOT_H_
+
+#include <iosfwd>
+
+namespace dali {
+namespace graph {
+
+class OpGraph;
+
+DLL_PUBLIC void GenerateDOTFromGraph(std::ostream &os, const OpGraph &graph,
```

Comment:
Changes compared to old `OpGraph::GenerateDOTFromGraph`
- show_ids removed
- ostream instead of ofstream (there was absolutely no reason to require a file stream)
- DOT header/footer is now printed inside this function (previously only the overload with a filename did that).